src/Files.App/ViewModels/UserControls/Widgets/RecentFilesWidgetViewModel.cs

```diff
@@ -265,13 +286,10 @@ private async Task ExecuteRemoveRecentItemCommand(RecentItem? item)
 		private async Task ExecuteClearRecentItemsCommand()
 		{
 			await _refreshRecentFilesSemaphore.WaitAsync();
+
 			try
 			{
-				Items.Clear();
-				bool success = App.RecentItemsManager.ClearRecentItems();
-
-				if (success)
-					IsEmptyRecentFilesTextVisible = true;
```

Comment:
is this done elsewhere now?
I think we could make this reactive to the `Items` collection or just restore this

Reply:
When you clear all, watcher triggers file change and update method does this. This was a kind of problem because even you fail to clear all it displays empty text.

> even you fail

Currently we seem to enumerate recent items and pinned files. SHAddToRecentDocs(null) works as emptying but this doesn't clear pinned files. I have another idea to solve but for now we keep what the main branch does